tests/integration/Tribe/Attendee_Registration/Theme_Template_Test.php

```diff
@@ -0,0 +1,349 @@
+				];
+			},
+		];
+
+		yield 'twentytwenty non-block theme, on AR page, on custom AR page' => [
+			function () {
+				switch_theme( 'twentytwenty' );
+
+				return [
+					'Twenty Twenty', // Theme name.
+					false, // Block theme status.
+					locate_template( 'singular.php' ), // Singular theme file.
+					true, // is_on_ar_page.
+					true, // is_on_custom_ar_page.
+					get_page_template(), // Expected template.
+				];
+			},
+		];
+
+		yield 'twentytwentyfour block theme, not on AR page' => [
+			function () {
+				switch_theme( 'twentytwentyfour' );
+
+				return [
+					'Twenty Twenty-Four', // Theme name.
+					true, // Block theme status.
+					'', // Singular theme file.
+					false, // is_on_ar_page.
+					false, // is_on_custom_ar_page (irrelevant here).
+					'', // Expected template.
+				];
+			},
+		];
+
+		yield 'twentytwentyfour block theme, on AR page, not on custom AR page' => [
+			function () {
+				switch_theme( 'twentytwentyfour' );
+
+				$template_slug = 'page'; // Default to 'page'.
+				$template      = get_template_directory() . DIRECTORY_SEPARATOR . $template_slug . '.html';
+
+				return [
+					'Twenty Twenty-Four', // Theme name.
+					true, // Block theme status.
+					'', // Singular theme file.
+					true, // is_on_ar_page.
+					false, // is_on_custom_ar_page.
+					locate_block_template( $template, 'page', [ $template_slug, 'page' ] ), // Expected template.
+				];
+			},
+		];
+
+		yield 'twentytwentyfour block theme, on AR page, on custom AR page' => [
+			function () {
+				switch_theme( 'twentytwentyfour' );
+
+				return [
+					'Twenty Twenty-Four', // Theme name.
+					true, // Block theme status.
+					'', // Singular theme file.
+					true, // is_on_ar_page.
+					true, // is_on_custom_ar_page.
+					get_page_template(), // Expected template.
+				];
+			},
+		];
+	}
+
+	/**
+	 * Test `setup_context` behavior with various scenarios.
+	 *
+	 * @test
+	 * @dataProvider setup_context_provider
+	 */
+	public function should_setup_context_correctly( Closure $fixture ): void {
+		[ $posts, $query, $is_main_query, $is_on_ar_page, $is_on_custom_ar_page, $expected ] = $fixture();
+
+		// Overwrite the `is_main_query` method using uopz.
+		$this->set_class_fn_return( WP_Query::class, 'is_main_query', $is_main_query );
+
+		// Mock `is_on_ar_page` and `is_on_custom_ar_page`.
+		$this->set_class_fn_return( Template::class, 'is_on_ar_page', $is_on_ar_page );
+		$this->set_class_fn_return( Template::class, 'is_on_custom_ar_page', $is_on_custom_ar_page );
+
+		// Create an instance of the class to test.
+		$template_instance = new Template();
+
+		// Run the method under test.
+		$result = $template_instance->setup_context( $posts, $query );
+
+		// Assert the result matches the expectation.
+		$this->assertEquals( $expected, $result, 'setup_context result should match the expected output.' );
+
+		// Delete the test post.
+		wp_delete_post( $posts[0]->ID, true );
```

Comment:
```suggestion
```

This should not be needed anymore. Since the posts are being created within a test case the DB's state remains unchanged. Thats because before every test case we have a "START TRANSACTION" statement and after each test case we have a "ROLLBACK" statements